packages/textlint-tester/src/textlint-tester.ts

```diff
@@ -71,47 +119,109 @@ export class TextLintTester {
         const ext = typeof valid === "object" && valid.ext !== undefined ? valid.ext : ".md";
         const textlint = new TextLintCore();
-        textlint.setupRules(
-            {
-                [ruleName]: rule
-            },
-            {
-                [ruleName]: options
+        if (isTestConfig(param)) {
+            const rules: any = {};
+            const rulesOptions: any = {};
+            param.rules.forEach(rule => {
+                const ruleName = rule.ruleId;
+                const ruleObject = rule.rule;
+                const ruleOptions = rule.options || {};
+                rules[ruleName] = ruleObject;
+                rulesOptions[ruleName] = ruleOptions;
+            });
+            textlint.setupRules(rules, rulesOptions);
+            if (param.plugins !== undefined) {
+                param.plugins.forEach(plugin => {
+                    const pluginName = plugin.pluginId;
+                    const pluginObject = plugin.plugin;
+                    const pluginOptions = plugin.options || {};
+                    textlint.setupPlugins(
+                        {
+                            [pluginName]: pluginObject
+                        },
+                        {
+                            [pluginName]: pluginOptions
+                        }
+                    );
+                });
             }
-        );
+        } else {
+            const options = (typeof valid === "object" && valid.options) || {};
+            textlint.setupRules(
+                {
+                    [name]: param
+                },
+                {
+                    [name]: options
+                }
+            );
+        }
         it(inputPath || text, () => {
             return testValid({ textlint, inputPath, text, ext });
         });
     }
 
-    testInvalidPattern(ruleName: string, rule: TextlintRuleCreator, invalid: TesterInvalid) {
+    testInvalidPattern(name: string, param: TextlintRuleCreator | TestConfig, invalid: TesterInvalid) {
         const errors = invalid.errors;
         const inputPath = invalid.inputPath;
         const text = invalid.text;
-        const options = invalid.options || {};
         const ext = invalid.ext !== undefined ? invalid.ext : ".md";
         const textlint = new TextLintCore();
-        textlint.setupRules(
-            {
-                [ruleName]: rule
-            },
-            {
-                [ruleName]: options
+        if (isTestConfig(param)) {
+            const rules: any = {};
+            const rulesOptions: any = {};
+            param.rules.forEach(rule => {
+                const ruleName = rule.ruleId;
+                const ruleObject = rule.rule;
+                const ruleOptions = rule.options || {};
+                rules[ruleName] = ruleObject;
+                rulesOptions[ruleName] = ruleOptions;
+            });
+            textlint.setupRules(rules, rulesOptions);
+            if (param.plugins !== undefined) {
```

Comment:
`Array.isArray(param.plugins)` is better.

Reply:
Alright. Fixed.